src/backend/tcop/pquery.c

```diff
@@ -394,10 +397,11 @@ ChoosePortalStrategy(List *stmts)
 					query->utilityStmt == NULL &&
 					query->parentStmtType == PARENTSTMTTYPE_NONE)
 				{
-					if (query->hasModifyingCTE)
-						return PORTAL_ONE_MOD_WITH;
-					else
+					if (!query->hasModifyingCTE)
```

Comment:
Current test cases cover only modifications from PlannedStmt branch.
Do we need tests for current branch (Query) or it's covered by some already existed tests?

Reply:
Have discovered the case when the Query is comes into the executor. 
```
PREPARE testplan (int, int) AS (with cte as (insert into t1 values ($1, $2) returning *) select * from cte);
```
Here, the `query->hasModifyingCte` is set to true

Reply:
So, can we add some tests to cover this block of changed code?

Reply:
Yes, we can. But to what sql section? I'm looking at prepare.sql and last test in there is from 2007

Reply:
But I feel we still use PlannedStmt in your example.

Reply:
Well, as I understand,  there is no difference for QD. The same portal strategy is choosen with our changes and whithout them (for QD the PrepareStmt is coming, after that the Query comes as well). And i doubt whether we need to test this branch. Probably we can even don't change the code for Query type, because to QE the PlannedStmt always comes.

Reply:
Removed the Query branch, because for QE we always expect the PlannedStmt

Reply:
After the voice chat, we decided to dig deeper into possibility of appearing of Query stmt.

Reply:
According to that [comment](https://github.com/arenadata/gpdb/blob/2de3b73a03045062849da9b99612bcd022334a7b/src/backend/tcop/postgres.c#L1127), the Query may come only with UTILITY command. One need to find an example of utility command, containing the modifying CTE.  After exploring references to this [function](https://github.com/arenadata/gpdb/blob/2de3b73a03045062849da9b99612bcd022334a7b/src/backend/cdb/dispatcher/cdbdisp_query.c#L411) across the code base, I did not manage to find an example, where we can pass the query with modifying CTE as utility command to QE.